src/plopp/backends/matplotlib/static.py

```diff
@@ -15,29 +15,44 @@ def _make_svg_repr(fig):
 REPR_MAP = {'png': _make_png_repr, 'svg': _make_svg_repr}
 
 
+def get_repr_maker(form=None, npoints=0):
+    """
+    Get the function used to create the mimebundle representation of a figure,
+    based on the format requested and the number of points in the figure.
+    """
+    if form is None:
+        form = 'png' if (npoints > 20_000) else 'svg'
+    return REPR_MAP[form.lower()]
+
+
 class StaticFig:
     """
     Create a static figure to represent one-dimensional data.
     """
 
     def __init__(self, *args, FigConstructor, **kwargs):
         self._fig = FigConstructor(*args, **kwargs)
+        self._args = args
+        self._kwargs = kwargs
+        self._fig_constructor = FigConstructor
```

Comment:
If this part is going to be repeated in other backend figure wrappers,
maybe we can extract them like this...?

```
class FigureWrapper:
    def __init_figure__(self, *args, FigConstructor, **kwargs):
        self._fig = FigConstructor(*args, **kwargs)
        self._args = args
        self._kwargs = kwargs
        self._fig_constructor = FigConstructor

class StaticFig(FigureWrapper):
    def __init__(self, *args, FigConstructor, **kwargs):
        self.__init_figure__(*args, FigConstructor, **kwargs)

class InteractiveFig(FigureWrapper, VBox):
    def __init__(self, *args, FigConstrutor, **kwargs):
        self.__init_figure__(*args, FigConstructor, **kwargs)
        # other VBox stuff
```

Reply:
I had issues in the past with multiple inheritance on `VBox`, but maybe it wasn't using a mixin.
I will try again, and maybe then I can clean up some other parts.

Reply:
It seems to work nicely, and I removed a bunch of duplication in the process :+1: